### PR TITLE
[CARBONDATA-3443] Update hive guide with Read from hive

### PR DESCRIPTION
**Pull request by dhatchayani**
Update hive-guide documentation with Read from Hive

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3836/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11903/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3630/

File: docs/hive-guide.md [hunk 1]
```diff
@@ -99,4 +99,16 @@ select count(*) from hive_carbon;
 select * from hive_carbon order by id;
 ```
 
+### Query data from hive table in Spark-2.1
+
+```
+set hive.mapred.supports.subdirectories=true;
+set mapreduce.input.fileinputformat.input.dir.recursive=true;
+
+create table hive_carbon_1(id int, name string, scale decimal, country string, salary double) ROW FORMAT SERDE 'org.apache.carbondata.hive.CarbonHiveSerDe' WITH SERDEPROPERTIES ('mapreduce.input.carboninputformat.databaseName'='default', 'mapreduce.input.carboninputformat.tableName'='HIVE_CARBON_EXAMPLE') STORED AS INPUTFORMAT 'org.apache.carbondata.hive.MapredCarbonInputFormat' OUTPUTFORMAT 'org.apache.carbondata.hive.MapredCarbonOutputFormat' LOCATION 'location_to_the_carbon_table';
```
Comment (chetandb) [inline docs/hive-guide.md]: "Example" can be the heading for this query sequence

File: docs/hive-guide.md [hunk 2]
```diff
@@ -99,4 +99,16 @@ select count(*) from hive_carbon;
...
+select * from hive_carbon_1;
+select count(*) from hive_carbon_1;
+select * from hive_carbon_1 order by id;
+```
+
```
Comment (chetandb) [inline docs/hive-guide.md]: query on hive table created using Carbon table with standard hive partition is not supported. This can be mentioned as a note.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3642/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11916/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3851/

File: docs/hive-guide.md [hunk 3]
```diff
@@ -83,20 +83,40 @@ copy snappy-java-xxx.jar from "./<SPARK_HOME>/jars/" to "./Library/Java/Extensio
...
+ - This is to read the carbon table through Hive. It is the integration of the carbon with Hive.
 
 ```
 set hive.mapred.supports.subdirectories=true;
-set mapreduce.input.fileinputformat.input.dir.recursive=true;
+set mapreduce.dir.recursive=true;
+These properties helps to recursively traverse through the directories to read the carbon folder structure.
+
+###Example
```
Comment (xubo245) [inline docs/hive-guide.md]: Please optimize it , View file

**Comment by xubo245** [comment]
@dhatchayani Can you add some picture for this guide? it will more easier for user to use hive and carbon

File: docs/hive-guide.md [hunk 4]
```diff
@@ -88,7 +88,7 @@ export HADOOP_OPTS="-Dorg.xerial.snappy.lib.path=/Library/Java/Extensions -Dorg.
 $HIVE_HOME/bin/hive
 ```
 
-### Query data from hive table
+### Query data from hive table in Spark versions above 2.1
```
[the remark this answers is not in the document]
Comment (dhatchayani) [reply docs/hive-guide.md]: that is removed already

**Comment by dhatchayani** [comment]
> @dhatchayani Can you add some picture for this guide? it will more easier for user to use hive and carbon

@xubo245 we have HiveExample class for example and this guide for the steps to be followed. I don't think we have added images to any guide document.

File: docs/hive-guide.md (hunk 3 above)
Comment (dhatchayani) [reply docs/hive-guide.md]: optimized

File: docs/hive-guide.md (hunk 2 above)
Comment (dhatchayani) [reply docs/hive-guide.md]: note added

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3696/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3908/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11971/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3699/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3911/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11974/

**Comment by dhatchayani** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3711/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3924/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11987/

**Comment by xubo245** [comment]
LGTM, thanks for your contribution!

**Pull request closed without merging**